### PR TITLE
"enable" typescript strictness

### PR DESCRIPTION
**Pull request by smadbe**
Enable typescript strictness ... but disabling some (almost all 😁) rules. The goal is to try to re-enable them one by one in the near future (some requires many changes). 

Also add linter rules embedded in the TS compiler itself.... but also disable some for now.

@SebastienTainon , if you have time, you could already fix in this PR the "noImplicitReturns" and "noPropertyAccessFromIndexSignature" rules which were not triggering many issues (and quite easy to fix).

### CODE REVIEW

**Comment by SebastienTainon** [comment]
@smadbe Shouldn't we setup aTS check in the Github checks on this PR?
It's fine to increase strictness but it seems to me that there is nothing to automatically check it currently.
Btw, what do you use to run a TS check on all the files? Running `yarn tsc` results in a JS heap of memory error on my side.

**Comment by smadbe** [comment]
The current build process (through `yarn`) does use the Typescript compiler? I was for sure it was tsc which was returning the errors...
Now I think about it... to test these rules, I was using `yarn dev`, and the errors were returned while I do not switch the flags to false, so I think the current build/dev process through `yarn` use the typescript computer under the hood. Yes, `yarn` uses `webpack` which calls `tsc`...
On this branch, change in `tsconfig.json` the property `noPropertyAccessFromIndexSignature` to true, you'll see the errors.

**Comment by SebastienTainon** [comment]
Yes you're right though it's a little more complicated than that: in the project I use the plugin https://www.npmjs.com/package/fork-ts-checker-webpack-plugin which is recommended to speed up development workflow. So when you do `yarn dev`, Typescript errors don't prevent you from building the project, however TS errors appear on the console in the forked process. But when `yarn build` is used, the plugin runs in build mode and seems to indeed block the build in case of errors, which I wasn't aware of. I though you needed to do `yarn tsc` to trigger manually the TS checker. That's good to know!

Btw I think my memory crash errors might be related to this, I've just added a commit to double the memory limit allowed to the TS checker.

**Review by SebastienTainon — APPROVED** [review]
I've fixed the two strict rules that you mentioned. It looks good to me!

File: tsconfig.json [hunk 1]
```diff
@@ -15,8 +14,6 @@
 
     "noFallthroughCasesInSwitch": true,
     "noUncheckedIndexedAccess": true,
-    "noPropertyAccessFromIndexSignature": false, // temporirly disabled, should be enabled (only 25 errors, easily fixable)
-    "noImplicitReturns": false, // temporirly disabled, should be enabled (only 5 errors, easily fixable)
```
Comment (smadbe) [inline tsconfig.json]: Unlike strict rules which are defaulted to true with the `"strict": true`, here their default values are `false`, so you should leave the rules to make sure they are checked at compile time.

Comment (SebastienTainon) [reply tsconfig.json]: Oops sorry, that's now fixed :)

**Pull request merged**